### ISSUE TITLE
[BUG] Handling view functions separately to executables (for typescript bindings)

### ISSUE THREAD

**Issue by starknetdev** [dojoengine/dojo#2720]
**Describe the bug**
When generating the `contracts.gen.ts` all functions defined are execute calls that return a transaction hash. Using an example of ERC20 `balance_of`, there is no easy way to get the result from this hash. They should really be generated as calls.

**To Reproduce**
* Clone [tournament_component](https://github.com/Provable-Games/arcade-components/tree/main)
* run, cd tournaments && cd contracts && sozo build --typescript
* inspect, contracts/bindings/typescript/contracts.gen.ts

or find the file here: [contracts.gen.ts](https://github.com/Provable-Games/arcade-components/tree/main/tournaments/contracts/bindings/typescript/contracts.gen.ts)

Using the `balance_of` example:

**Current behavior**
```
	const erc20_mock_balanceOf = async (snAccount: Account, account: string) => {
		try {
			return await provider.execute(
				snAccount,
				{
					contractName: "erc20_mock",
					entrypoint: "balance_of",
					calldata: [account],
				},
				"tournament",
			);
		} catch (error) {
			console.error(error);
		}
	};
```

**Expected behavior**

```
  const erc20_mock_balance_of = async (account: Account, address: string) => {
    try {
      return await provider.call("tournament", {
        contractName: "erc20_mock",
        entrypoint: "balance_of",
        calldata: [address],
      });
    } catch (error) {
      console.error(error);
    }
  };
  ```

**Additional context**
From discussing with Glihm the best approach seems to be having an explicit way in Dojo of defining a view function, opposed to executable functions. Just trying to detect if a function has a return type isn't optimal as it can be useful for executables to have these (intra-contract and contract-contract calls).

**Comment by MartianGreed** [comment]
ohayo ! I've fixed it here #2725 .

In `sozo build` command, we use the `gather_dojo_data` function that goes over Abi to fetch out different tokens, so in fact I was able to use `StateMutability` from Abi

**Comment by glihm** [comment]
Merged in #2725, but will be available in the next patch version.
